compiler/resolution/implementForallIntents.cpp

```diff
@@ -920,6 +947,96 @@ static void resolveAndPruneExplicitShadowVars(ForallStmt* fs,
   }
 }
 
+// If it is a CallExpr(<something>, _mt, arg), and <something>
+// names a field in 'recType', return that field's Symbol
+static Symbol* isFieldAccess(AggregateType* recType, Expr* arg) {
+  if (CallExpr* call = toCallExpr(arg->parentExpr))
+    if (call->numActuals() == 2)
+      if (call->get(1)->getValType() == dtMethodToken)
+        // Field accesses are unresolved method calls at this point.
+        if (UnresolvedSymExpr* base = toUnresolvedSymExpr(call->baseExpr))
+          if (Symbol* fieldSym = recType->getField(base->unresolved, false))
```

Comment:
Hmm... Looking for a field with the same name and assuming that the UnresolvedSymExpr would resolve to it seems a bit fraught to me. Nonetheless I think it is OK for now. It might be worth putting a comment that it might not be sufficient in the future (is that right? could one create parentheses-less methods that shadow the field, say?)